x-pack/plugins/observability_solution/slo/server/lib/rules/slo_burn_rate/lib/should_suppress_instance_id.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ALL_VALUE } from '@kbn/slo-schema';
+import { EvaulateDependenciesResponse } from './evaluate_dependencies';
+
+export function shouldSuppressInstanceId(
+  results: EvaulateDependenciesResponse['activeRules'],
+  instanceId: string
+) {
+  return results.reduce((acc, res) => {
+    if (acc === true) return acc;
+    if (res.suppressAll === true) return true;
+    if (instanceId === ALL_VALUE && res.instanceIdsToSuppress.length > 0) return true;
+    return (
+      res.instanceIdsToSuppress.length > 0 &&
+      res.instanceIdsToSuppress.some((id) => id === instanceId)
```

Comment:
If I understand correctly, we're trying to match the parents instanceId to the dependency's instance id.

Unfortunately, our new multi-group by and changes to the way instanceId is generated will create conflicts for this. All SLOs, parent and dependencies, will need to be updated to the latest spec in order to use this feature.

Reply:
Crap! With the new multi-group-by, the fingerprints (and the previous `key:value` strings) will only match if the field names and values are exactly the same. For example, lets say I did a group by on `labels.project_id` for the parent and the dependency was grouped on `labels.projectId`, the fingerprints will be different even though the values of `labels.projectId` and `labels.project_id` are the same. Either way, when I wrote this implementation, it was based on a single group by where we stored only the value in the `slo.instanceId`. 

I'm not sure we can ship this till we figure out how to resolve the instances OR the hard requirement will be that the field names and values must be exact.